README.md

```diff
@@ -10,12 +10,328 @@ A core widget library for Dojo 2.
+    }
+});
+
+export default createLabelWidget;
+```
+With its usages as follows:
+
+```ts
+import createLabelWidget from './widgets/createLabelWidget';
+
+const label = createLabelWidget({ state: { label: 'I am a label' }});
+```
+
+To create structured widgets override the `getChildrenNodes` function.
+
+```ts
+import { ComposeFactory } from 'dojo-compose/compose';
+import { DNode, Widget, WidgetOptions, WidgetState } from 'dojo-interfaces/widgetBases';
+import createWidgetBase from 'dojo-widgets/bases/createWidgetBase';
+import d from 'dojo-widgets/util/d';
+
+interface ListItem {
+    name: string;
+}
+
+interface ListState extends WidgetState {
+    items?: ListItem[];
+};
+
+interface ListOptions extends WidgetOptions<ListState> { };
+
+type List = Widget<ListState>;
+
+interface ListFactory extends ComposeFactory<List, ListOptions> { };
+
+function isEven(value: number) {
+    return value % 2 === 0;
+}
+
+function listItem(item: ListItem, itemNumber: number): DNode {
+    const classes = isEven(itemNumber) ? {} : { 'odd-row': true };
+    return d('li', { innerHTML: item.name, classes });
+}
+
+const createListWidget: ListFactory = createWidgetBase.mixin({
+    mixin: {
+        getChildrenNodes: function (this: List): DNode[] {
+            const listItems = this.state.items.map(listItem);
+
+            return [ d('ul', {}, listItems) ];
+        }
+    }
+});
+
+export default createListWidget;
+```
+
+### Projector
+
+To render widgets they must be appended to a `projector`. It is possible to create many projectors and attach them to `Elements` in the `DOM`, however `projectors` must not be nested.
+
+### Dojo Widget Components
+
+A selection of core reusable widgets are provided for convenience that are fully accessible and internationalizable.
+
+To be completed, list core components here.
+
+## How do I contribute?
+
+We appreciate your interest!  Please see the [Dojo 2 Meta Repository](https://github.com/dojo/meta#readme) for the
+Contributing Guidelines and Style Guide.
+
+### Installation
+
+To start working with this package, clone the repository and run `npm install`.
+
+In order to build the project run `grunt dev` or `grunt dist`.
+
+### Testing
+
+Test cases MUST be written using [Intern](https://theintern.github.io) using the Object test interface and Assert assertion interface.
+
+90% branch coverage MUST be provided for all code submitted to this repository, as reported by istanbul’s combined coverage results for all supported platforms.
+
+To test locally in node run:
+
+`grunt test`
+
+To test against browsers with a local selenium server run:
+
+`grunt test:local`
+
+To test against BrowserStack or Sauce Labs run:
+
+`grunt test:browserstack`
+
+or
+
+`grunt test:saucelabs`
 
-© 2016 Dojo Foundation & contributors. [New BSD](http://opensource.org/licenses/BSD-3-Clause) license.
+## Licensing information
 
+© 2016 Dojo Foundation & contributors. [New BSD](http://opensource.org/licenses/BSD-3-Clause) license.
```

Comment:
Should be JS Foundation rather than Dojo Foundation, and perhaps link to https://js.foundation/ .